swap-logic/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+#![cfg_attr(not(test), no_std)]
+
+mod constants;
+mod swap_control;
+
+extern crate alloc;
+
+use alloc::{string::String, vec::Vec};
+
+use crate::constants::methods;
+use contract::{contract_api::runtime, unwrap_or_revert::UnwrapOrRevert};
+use types::{account::PublicKey, ApiError, CLValue, Key, U512};
+
+#[no_mangle]
+pub extern "C" fn delegate() {
+    let method_name: String = runtime::get_arg(0)
+        .unwrap_or_revert_with(ApiError::MissingArgument)
+        .unwrap_or_revert_with(ApiError::InvalidArgument);
+
+    match method_name.as_str() {
+        methods::METHOD_SET_SWAP_HASH => {
```

Comment:
`SWAH_HASH` has already been saved from swap_install to admin's named_key.
It seems to be a function you probably don't need.

Reply:
https://github.com/hdac-io/swap/pull/1/commits/28efabcc9e3ca17e99262555548a28195b956f5a